x-pack/plugins/ml/server/routes/trained_models.ts

```diff
@@ -155,6 +184,44 @@ export function trainedModelsRoutes({ router, routeGuard }: RouteInitialization)
     })
   );
 
+  /**
+   * @apiGroup TrainedModels
+   *
+   * @api {put} /api/ml/trained_models/:modelId Put a trained model
+   * @apiName PutTrainedModel
+   * @apiDescription Adds a new trained model
+   */
+  router.put(
+    {
+      path: '/api/ml/trained_models/{modelId}',
+      validate: {
+        params: modelIdSchema,
+        body: schema.any(),
```

Comment:
I really don't think we should be validating the shape of the trained models for endpoints which are just wrappers for es endpoints.
es will instantly complain if the model JSON is incorrect.

I know we do this validation on AD and DFA endpoints, but I think we should consider the idea of removing these validations too.
This is something to discuss outside of this PR.